qa/tasks/cephfs/test_volumes.py

```diff
@@ -311,6 +311,12 @@ def _wait_for_subvol_trash_empty(self, subvol, group="_nogroup", timeout=30):
             else:
                 raise
 
+    def _set_config_options_back_to_default(self):
```

Comment:
This should not be necessary. See `CephTestCase.tearDown` and `.config_clear`.

Reply:
We encountered this issue while testing https://github.com/ceph/ceph/pull/52670. We didn't get the config options cleared within a test class. Here's the link: https://github.com/ceph/ceph/pull/52670#discussion_r1484152748

Reply:
Sadly, these are maintained inside mgr/volumes. Is there any way to move it to src/common/options yaml ?

Reply:
@neesingh-rh Is there a link to a failed run that prompted this change?

Reply:
Here's the failure link: https://github.com/ceph/ceph/pull/52670#pullrequestreview-1841436109

Reply:
@neesingh-rh That failure was reported on Jan 24 and the PR was merged on Feb 20 - are you saying that the failures wasn't RCA'd and the change was approved and merged (the tests passed after you pushed in some fixes)?

Reply:
The issue we faced for our particular tests got fixed by reverting the configs to the default at the end of the tests which were part of  this particular https://github.com/ceph/ceph/pull/52670 PR. But https://github.com/ceph/ceph/pull/57627 fix adds a function which disables all config options at the end of every tests from now-onwards, so that we don't face this issue again in future.
See this comment where it is advised to revert the configs at the end of every tests: https://github.com/ceph/ceph/pull/52670#discussion_r1484152748

Reply:
In that entire run, there is just one invocation of `config rm` for `snapshot_clone_no_wait`.

```
zgrep "config rm" teuthology.log.gz  | grep snapshot_clone_no_wait
2024-01-22T08:27:52.269 DEBUG:teuthology.orchestra.run.smithi031:> sudo adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 120 ceph --cluster ceph config rm mgr mgr/volumes/snapshot_clone_no_wait
2024-01-22T08:27:52.620 INFO:teuthology.orchestra.run.smithi031.stderr:2024-01-22T08:27:52.620+0000 7f4fef20f700  1 -- 172.21.15.31:0/3759669512 --> [v2:172.21.15.31:3300/0,v1:172.21.15.31:6789/0] -- mon_command({"prefix": "config rm", "who": "mgr", "name": "mgr/volumes/snapshot_clone_no_wait"} v 0) v1 -- 0x7f4fe8068230 con 0x7f4fe80d0920
2024-01-22T08:27:52.623 INFO:teuthology.orchestra.run.smithi031.stderr:2024-01-22T08:27:52.624+0000 7f4fe57fa700  1 -- 172.21.15.31:0/3759669512 <== mon.0 v2:172.21.15.31:3300/0 7 ==== mon_command_ack([{"prefix": "config rm", "who": "mgr", "name": "mgr/volumes/snapshot_clone_no_wait"}]=0  v73) v1 ==== 117+0+0 (secure 0 0 0) 0x7f4fe0059c50 con 0x7f4fe80d0920
```

@neesingh-rh Suggest debugging into why that happened - we should have seen that config removed for tests that did a `config set` for the config (which seems to be a handful of tests).

Reply:
> In that entire run, there is just one invocation of `config rm` for `snapshot_clone_no_wait`.
> 
> ```
> zgrep "config rm" teuthology.log.gz  | grep snapshot_clone_no_wait
> 2024-01-22T08:27:52.269 DEBUG:teuthology.orchestra.run.smithi031:> sudo adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 120 ceph --cluster ceph config rm mgr mgr/volumes/snapshot_clone_no_wait
> 2024-01-22T08:27:52.620 INFO:teuthology.orchestra.run.smithi031.stderr:2024-01-22T08:27:52.620+0000 7f4fef20f700  1 -- 172.21.15.31:0/3759669512 --> [v2:172.21.15.31:3300/0,v1:172.21.15.31:6789/0] -- mon_command({"prefix": "config rm", "who": "mgr", "name": "mgr/volumes/snapshot_clone_no_wait"} v 0) v1 -- 0x7f4fe8068230 con 0x7f4fe80d0920
> 2024-01-22T08:27:52.623 INFO:teuthology.orchestra.run.smithi031.stderr:2024-01-22T08:27:52.624+0000 7f4fe57fa700  1 -- 172.21.15.31:0/3759669512 <== mon.0 v2:172.21.15.31:3300/0 7 ==== mon_command_ack([{"prefix": "config rm", "who": "mgr", "name": "mgr/volumes/snapshot_clone_no_wait"}]=0  v73) v1 ==== 117+0+0 (secure 0 0 0) 0x7f4fe0059c50 con 0x7f4fe80d0920
> ```
> 
> @neesingh-rh Suggest debugging into why that happened - we should have seen that config removed for tests that did a `config set` for the config (which seems to be a handful of tests).

I am on it today.

Reply:
> In that entire run, there is just one invocation of `config rm` for `snapshot_clone_no_wait`.
> 
> ```
> zgrep "config rm" teuthology.log.gz  | grep snapshot_clone_no_wait
> 2024-01-22T08:27:52.269 DEBUG:teuthology.orchestra.run.smithi031:> sudo adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 120 ceph --cluster ceph config rm mgr mgr/volumes/snapshot_clone_no_wait
> 2024-01-22T08:27:52.620 INFO:teuthology.orchestra.run.smithi031.stderr:2024-01-22T08:27:52.620+0000 7f4fef20f700  1 -- 172.21.15.31:0/3759669512 --> [v2:172.21.15.31:3300/0,v1:172.21.15.31:6789/0] -- mon_command({"prefix": "config rm", "who": "mgr", "name": "mgr/volumes/snapshot_clone_no_wait"} v 0) v1 -- 0x7f4fe8068230 con 0x7f4fe80d0920
> 2024-01-22T08:27:52.623 INFO:teuthology.orchestra.run.smithi031.stderr:2024-01-22T08:27:52.624+0000 7f4fe57fa700  1 -- 172.21.15.31:0/3759669512 <== mon.0 v2:172.21.15.31:3300/0 7 ==== mon_command_ack([{"prefix": "config rm", "who": "mgr", "name": "mgr/volumes/snapshot_clone_no_wait"}]=0  v73) v1 ==== 117+0+0 (secure 0 0 0) 0x7f4fe0059c50 con 0x7f4fe80d0920
> ```
> 
> @neesingh-rh Suggest debugging into why that happened - we should have seen that config removed for tests that did a `config set` for the config (which seems to be a handful of tests).

After spending time with the changes in PR and logs, I found that one invocation of `config rm` for `snapshot_clone_no_wait` is expected. As some additions of `config_set` for `snapshot_clone_no_wait` were done after that particular run and to be precise after this PR got merged. See this: https://github.com/ceph/ceph/pull/55660 .
And `test_subvolume_snapshot_clone_when_no_wait_enabled` fails and only one test i.e. `test_subvolume_snapshot_clone_ cancel_pending` containing `config_set` for `snapshot_clone_no_wait` succeeds for which we saw `config rm` .

Reply:
I checked working of `.config_clear` through vstart_runner as well as teuthology. It seems it works perfectly and c`config rm` happens after each and every particular tests.